packages/twenty-server/src/database/commands/upgrade-version-command/0-54/0-54-created-by-default-value.command.ts

```diff
@@ -0,0 +1,100 @@
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { Command } from 'nest-commander';
+import { Repository } from 'typeorm';
+import chalk from 'chalk';
+import { FieldMetadataType } from 'twenty-shared/types';
+
+import {
+  ActiveOrSuspendedWorkspacesMigrationCommandRunner,
+  RunOnWorkspaceArgs,
+} from 'src/database/commands/command-runners/active-or-suspended-workspaces-migration.command-runner';
+import { Workspace } from 'src/engine/core-modules/workspace/workspace.entity';
+import { TwentyORMGlobalManager } from 'src/engine/twenty-orm/twenty-orm-global.manager';
+import { ObjectMetadataEntity } from 'src/engine/metadata-modules/object-metadata/object-metadata.entity';
+import { ActorMetadata } from 'src/engine/metadata-modules/field-metadata/composite-types/actor.composite-type';
+import { generateDefaultValue } from 'src/engine/metadata-modules/field-metadata/utils/generate-default-value';
+import { WorkspaceDataSourceService } from 'src/engine/workspace-datasource/workspace-datasource.service';
+import { computeTableName } from 'src/engine/utils/compute-table-name.util';
+
+@Command({
+  name: 'upgrade:0-54:0-54-created-by-default-value',
+  description: 'Fix createdBy default value',
+})
+export class FixCreatedByDefaultValueCommand extends ActiveOrSuspendedWorkspacesMigrationCommandRunner {
+  constructor(
+    @InjectRepository(Workspace, 'core')
+    protected readonly workspaceRepository: Repository<Workspace>,
+    protected readonly twentyORMGlobalManager: TwentyORMGlobalManager,
+    @InjectRepository(ObjectMetadataEntity, 'metadata')
+    private readonly objectMetadataRepository: Repository<ObjectMetadataEntity>,
+    private readonly workspaceDataSourceService: WorkspaceDataSourceService,
+  ) {
+    super(workspaceRepository, twentyORMGlobalManager);
+  }
+
+  override async runOnWorkspace({
+    workspaceId,
+    options,
+  }: RunOnWorkspaceArgs): Promise<void> {
+    const objectsMetadata = await this.objectMetadataRepository.find({
+      where: { workspaceId },
+      relations: ['fields'],
+    });
+
+    for (const objectMetadata of objectsMetadata) {
+      const createdByFieldExists = objectMetadata.fields.some(
+        (field) => field.name === 'createdBy',
+      );
+
+      if (!createdByFieldExists) {
+        continue;
+      }
+
+      const workspaceDataSource =
+        await this.twentyORMGlobalManager.getDataSourceForWorkspace({
```

Comment:
use dataSource that is available in function parameter, no need to instantiate a new one:ActiveOrSuspendedWorkspacesMigrationCommandRunner is already providing one and destroying it properly to avoid memory leak)